lib/rubocop/cop/internal_affairs/redundant_message_argument.rb

```diff
@@ -10,39 +10,61 @@ module InternalAffairs
       # @example
       #
       #   # bad
-      #   add_offense(node, :expression, MSG)
-      #   add_offense(node, :expression, message)
-      #   add_offense(node, :expression, message(node))
+      #   add_offense(node, message: MSG)
+      #   add_offense(node, message: message)
+      #   add_offense(node, message: message(node))
       #
       #   # good
-      #   add_offense(node, :expression)
-      #   add_offense(node, :expression, CUSTOM_MSG)
-      #   add_offense(node, :expression, message(other_node))
+      #   add_offense(node)
+      #   add_offense(node, message: CUSTOM_MSG)
+      #   add_offense(node, message: message(other_node))
       #
       class RedundantMessageArgument < Cop
         MSG = 'Redundant message argument to `#add_offense`.'.freeze
 
         def_node_matcher :node_type_check, <<-PATTERN
-          (send nil :add_offense _offender _
-            {(const nil :MSG) (send nil :message) (send nil :message _offender)})
+          (send nil :add_offense $_node $hash)
         PATTERN
 
+        def_node_matcher :offending_message_argument, <<-PATTERN
```

Comment:
Changing `offending` to `redundant` would provide some hint to *why* it is offending.